packages/repository-json-schema/src/__tests__/unit/filter-json-schema.unit.ts

```diff
@@ -127,6 +129,50 @@ describe('getFilterJsonSchemaFor', () => {
     ]);
   });
 
+  it('allows free-form properties in "fields" for non-strict models"', () => {
+    const filter = {fields: ['test', 'id']};
+    ajv.validate(dynamicCustomerFilterSchema, filter);
+    expect(ajv.errors ?? []).to.be.empty();
+  });
+
+  it('allows only defined properties in "fields" for strict models"', () => {
+    const filter = {fields: ['test']};
+    ajv.validate(customerFilterSchema, filter);
+    expect(ajv.errors ?? []).to.containDeep([
+      {
+        keyword: 'enum',
+        dataPath: '.fields[0]',
+        message: 'should be equal to one of the allowed values',
```

Comment:
nitpick: 
> should be equal to one of the allowed values

Does error msg contain more details of the allowed values?

Reply:
```ts
{
    keyword: 'enum',
    dataPath: '.fields[0]',
    schemaPath: '#/properties/fields/oneOf/1/items/enum',
    params: { allowedValues: [ 'id', 'name' ] },
    message: 'should be equal to one of the allowed values'
  }
```
this is the object